ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/AggregateAnalyzer.java

```diff
@@ -276,17 +273,31 @@ private void enforceAggregateRules() {
     }
   }
 
+  /**
+   * This visitor performs two tasks: Create the input schema to the AggregateNode and validations.
+   *
+   * For creating the input schema, it checks if any expression along the path from the root
+   * expression to the leaf (UnqualifiedColumnReference) is part of the groupBy. If at least one is,
+   * then the UnqualifiedColumnReference is added to the schema.
+   *
+   * For validation, the visitor checks that:
+   * 1) expressions in non-aggregate functions are part of the grouping clause,
+   * 2) aggregate functions are not nested
+   * 3) window clauses (windowstart, windowend) don't appear in aggregate functions or groupBy
+   * 4) aggregate functions don't appear in the groupBy clause
+   * 5) expressions in the having clause are either aggregate functions or grouping keys
+   */
```

Comment:
first, yay for comments! Thanks for writing this detailed comment, helps future people (including me when I forget what this does) understand it 🎉 

second, nit: i'm pretty sure checkstyle forces the following format, but if it doesn't it'll make it easier to read anyway :D 

```suggestion
  /**
   * This visitor performs two tasks: Create the input schema to the AggregateNode and validations.
   *
   * <p> For creating the input schema, it checks if any expression along the path from the root
   * expression to the leaf (UnqualifiedColumnReference) is part of the groupBy. If at least one is,
   * then the UnqualifiedColumnReference is added to the schema.
   *
   * <p>For validation, the visitor checks that:
   * <ol>
   *   <li> expressions in non-aggregate functions are part of the grouping clause </li>
   *   <li> aggregate functions are not nested </li>
   *   <li> window clauses (windowstart, windowend) don't appear in aggregate functions or groupBy </li>
   *   <li> aggregate functions don't appear in the groupBy clause </li>
   *   <li> expressions in the having clause are either aggregate functions or grouping keys </li>
   * </ol>
   */
```